File: backend/Models/UserTournament.cs
namespace backend.Models
{
    public class UserTournament
    {
        public int UserId { get; set; }
        public User User { get; set; }

        public int TournamentId { get; set; }
        public Tournament Tournament { get; set; }

        public PaymentStatus PaymentStatus { get; set; }
        public DateTime? JoinedAt { get; set; }
    }
}

File: frontend/src/shared/types/models/userTournament.ts
import { TournamentDto } from "./tournament";

export interface UsersTournaments {
    tournament: TournamentDto,
    paymentStatus: string,
    joinedAt: string,
}

export interface JoinTournamentData {
    userId: number,
    tournamentId: number,
    isPaid: boolean,
}

export interface LeaveTournamentData {
    userId: number,
    tournamentId: number,
}